api/v1alpha1/stresschaos_types.go

```diff
@@ -0,0 +1,275 @@
+type StressChaosSpec struct {
+	// Mode defines the mode to run chaos action.
+	// Supported mode: one / all / fixed / fixed-percent / random-max-percent
+	Mode PodMode `json:"mode"`
+
+	// Value is required when the mode is set to `FixedPodMode` / `FixedPercentPodMod` / `RandomMaxPercentPodMod`.
+	// If `FixedPodMode`, provide an integer of pods to do chaos action.
+	// If `FixedPercentPodMod`, provide a number from 0-100 to specify the max % of pods the server can do chaos action.
+	// If `RandomMaxPercentPodMod`,  provide a number from 0-100 to specify the % of pods to do chaos action
+	// +optional
+	Value string `json:"value"`
+
+	// Selector is used to select pods that are used to inject chaos action.
+	Selector SelectorSpec `json:"selector"`
+
+	// Stressors defines plenty of stressors supported to stress system components out.
+	// You can use one or more of them to make up various kinds of stresses. At least
+	// one of the stressors should be specified.
+	// +optional
+	Stressors *Stressors `json:"stressors"`
+
+	// StressngStressors defines plenty of stressors just like `Stressors` except that it's an experimental
+	// feature and more powerful. You can define stressors in `stress-ng` (see also `man stress-ng`) dialect,
+	// however not all of the supported stressors are well tested. It maybe retired in later releases. You
+	// should always use `Stressors` to define the stressors and use this only when you want more stressors
+	// unsupported by `Stressors`. When both `StressngStressors` and `Stressors` are defined, `StressngStressors`
+	// wins.
+	// +optional
+	StressngStressors string `json:"stressngStressors,omitempty"`
+
+	// Duration represents the duration of the chaos action
+	// +optional
+	Duration *string `json:"duration,omitempty"`
+
+	// Scheduler defines some schedule rules to control the running time of the chaos experiment about time.
+	// +optional
+	Scheduler *SchedulerSpec `json:"scheduler,omitempty"`
+
+	// Next time when this action will be applied again
+	// +optional
+	NextStart *metav1.Time `json:"nextStart,omitempty"`
+
+	// Next time when this action will be recovered
+	// +optional
+	NextRecover *metav1.Time `json:"nextRecover,omitempty"`
+
+	// If this chaos has been paused
+	// +optional
+	Paused bool `json:"paused"`
+}
+
+// GetSelector is a getter for Selector (for implementing SelectSpec)
+func (in *StressChaosSpec) GetSelector() SelectorSpec {
+	return in.Selector
+}
+
+// GetMode is a getter for Mode (for implementing SelectSpec)
+func (in *StressChaosSpec) GetMode() PodMode {
+	return in.Mode
+}
+
+// GetValue is a getter for Value (for implementing SelectSpec)
+func (in *StressChaosSpec) GetValue() string {
+	return in.Value
+}
+
+// StressChaosStatus defines the observed state of StressChaos
+type StressChaosStatus struct {
+	ChaosStatus `json:",inline"`
+	// Instances always specifies stressing instances
+	// +optional
+	Instances map[types.UID]StressInstance `json:"instances,omitempty"`
```

Comment:
I hope we can use a formatted string as the key, such as `namespace/podname`, instead of just the uid of the pod. That should be more readable. The current is like below, `91cb8ac8-a123-4568-a180-80734d3533fd` doesn't mean anything to user

``` yaml
    instances:
      91cb8ac8-a123-4568-a180-80734d3533fd:
        startTime: 1587763257000
        uid: "7346"
```